packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -116,25 +187,31 @@ type ContextMenuContentPrimitive = Polymorphic.ForwardRefComponent<
 
 const ContextMenuContent = React.forwardRef((props, forwardedRef) => {
   const { disableOutsidePointerEvents = true, offset, ...contentProps } = props;
+  const isSubmenu = React.useContext(SubmenuContext);
   const context = useContextMenuContext(CONTENT_NAME);
+
   return (
-    <MenuPrimitive.Content
-      {...contentProps}
-      ref={forwardedRef}
-      disableOutsidePointerEvents={context.open ? disableOutsidePointerEvents : false}
-      style={{
-        ...props.style,
-        // re-namespace exposed content custom property
-        ['--radix-context-menu-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
-      }}
-      trapFocus
-      disableOutsideScroll
-      portalled
-      side="bottom"
-      sideOffset={offset}
-      align="start"
-      alignOffset={2}
-    />
+    <InsideContentContext.Provider value={true}>
+      <MenuPrimitive.Content
+        {...contentProps}
+        {...(!isSubmenu && {
+          disableOutsidePointerEvents: context.open ? disableOutsidePointerEvents : false,
+          trapFocus: true,
+          disableOutsideScroll: true,
+          portalled: true,
+          side: 'bottom' as const,
+          align: 'start' as const,
+          alignOffset: 2,
+        })}
```

Comment:
I don't love the inline conditional spread like that, I find it a bit hard to read or even see that this is about the submenu.
I wonder if we'd be better off doing something like:

```tsx
const subMenuContentProps = isSubmenu ? { … } : {};
…
<MenuPrimitive.Content
  {...contentProps}
  {...submenuProps}
   …
/>
```

Not great either though. Anybody got other ideas to make this clearer?
Or we could duplicate the whole jsx block with a big ternary around `isSubmenu`.
I'm actually wondering if this means we did the inner swap too high up in `Menu`.

Reply:
I tripped up on this one too but at the time didn't have any better ideas. Just thinking now though, maybe we could do this (which I did for `ScrollArea`):

```jsx
const commonProps = { ...contentProps, sideOffset: offset, style: { ... } }

return isSubMenu ? (
  <MenuPrimitive.Content
    {...commonProps}
    ref={forwardedRef}
    disableOutsidePointerEvents={context.open ? disableOutsidePointerEvents : false}
    trapFocus={true}
    disableOutsideScroll={true}
    portalled={true}
    side="bottom"
    align="start"
    alignOffset={2}
  />
) : (
  <MenuPrimitive.Content {...commonProps} ref={forwardedRef} />
);
```

Reply:
Yep, I like that much better. Great idea! 👍

Reply:
Love it 👍